sklearn/metrics/pairwise.py

```diff
@@ -53,13 +53,14 @@ def _return_float_dtype(X, Y):
     1. If dtype of X and Y is float32, then dtype float32 is returned.
     2. Else dtype float is returned.
     """
+    xp, _ = get_namespace(X, Y)
     if not issparse(X) and not isinstance(X, np.ndarray):
-        X = np.asarray(X)
+        X = xp.asarray(X)
 
     if Y is None:
         Y_dtype = X.dtype
     elif not issparse(Y) and not isinstance(Y, np.ndarray):
-        Y = np.asarray(Y)
+        Y = xp.asarray(Y)
```

Comment:
I have the feeling that this line is not needed. Can you try to remove it? Then we can simplify the if / else structure of this code.